### PR TITLE
chore: update cluster address for react preview [DET-5008]

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

updates the default server address set for the react preview build to a cors proxy served over https

## Test Plan

- have no running determined clusters on your machine and visit `build-react-preview` artifact from circle-ci on this pr and preview the UI changes

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1035,7 +1035,7 @@ jobs:
       - image: cimg/node:12.16
     environment:
       PUBLIC_URL: /0
-      SERVER_ADDRESS: http://localhost:8080
+      SERVER_ADDRESS: https://detai.ml/proxy
```
Comment (hkang1) [inline .circleci/config.yml:1038]: **question:** I'm a bit clueless here, but where is this URL from?

Comment (hkang1) [reply .circleci/config.yml:1038]: **suggestion:** would it help to have a comment above this explaining that it allows https connectivity and that it points to `latest-master.determined.at:8080`?

**Review by hkang1 — APPROVED** [review]
Pretty cool to see CI previews against latest-master!

**Comment by hamidzr** [comment]
@dzhu any thoughts here?

**Review by dzhu — COMMENTED** [review]
Uh, I don't know anything about what's up that URL or what SERVER_ADDRESS is used for here, so... looks good? Though I'll note that everything under that domain seems to be returning 502s, which seems weird.

**Comment by hamidzr** [comment]
> Uh, I don't know anything about what's up that URL or what SERVER_ADDRESS is used for here, so... looks good? Though I'll note that everything under that domain seems to be returning 502s, which seems weird.

let's chat over slack more about SERVER_ADDRESS in `webui/react/README.md`

**Pull request merged**